src/audio/volume.c

```diff
@@ -191,13 +191,14 @@ static struct comp_dev *volume_new(struct sof_ipc_comp *comp)
 
 	comp_set_drvdata(dev, cd);
 	work_init(&cd->volwork, vol_work, dev, WORK_ASYNC);
-
+	/* set volume min/max levels */
+	vol_set_min_max_levels(cd, ipc_vol->min_value, ipc_vol->max_value);
 	/* set the default volumes */
 	for (i = 0; i < PLATFORM_MAX_CHANNELS; i++) {
-		cd->volume[i] = VOL_ZERO_DB;
-		cd->tvolume[i] = VOL_ZERO_DB;
+		cd->volume[i] = cd->max_volume;
```

Comment:
I think the intention is that default volume should be 0dB during init and this was updated recently too. I've added Seppo to comment.

Reply:
@mrajwa looks like you have not committed, squashed or pushed your update ? All I can see here is adding two empty lines ?

Reply:
What issue is addressed with this patch?

I don't think it's a good idea to initialize volume to maximum. Max is not necessarily 0 dB (neutral), it could be e.g. +20 dB. We will support a larger volume range in future and this code part would need to change.

I added originally somewhere in SOF 1.1 freezing the vol max check due to generic C volume implementation issue where some volume conversions would overflow above the defined max. But those overflows should be treated as bugs now. There should be no PCM samples overflows with any 32 bit gain value but predictable saturation.

Reply:
One more comment -- Though in the case if volume max is less than 0dB then you should initialize to max from ipc. So,

cd->tvolume[i] = MIN(cd->max_volume, VOL_ZERO_DB);
cd->volume[i] = cd->tvolume[i];

Right?

Reply:
Okay, I think at this point it would be better if we organize a short meeting to discuss it all. @singalsu would you be open for it?